### PR TITLE
Bump hsqldb to v2.7.1

### PR DESCRIPTION
**Pull request by pmauduit**
details of the CVE here: https://nvd.nist.gov/vuln/detail/CVE-2022-41853

Tests: `mvn verify` OK locally

# Checklist

- [ ] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [ ] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [ ] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [ ] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [ ] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

**Comment by pmauduit** [comment]
hsqldb 2.7.1 is compiled against java 11 (bytecode version 55.0), I haven't paid attention to this, as I was locally using a jdk >= 11.

File: pom.xml [hunk 1]
```diff
@@ -1030,7 +1030,7 @@
       <dependency>
         <groupId>org.hsqldb</groupId>
         <artifactId>hsqldb</artifactId>
-        <version>2.4.1</version>
+        <version>2.7.1</version>
```
Comment (mprins) [inline pom.xml:1033]: This is a Java 11 (class file version 55.0) build so will cause failures when used with any Java version before 11.
Most of the CI is still running on Java 8

**Comment by aaime** [comment]
GeoTools main branch (development) is going to switch to Java 11 as the minimum soon, but the stable and maintenance branches will keep supporting Java 8. If the CVE is actually actionable, I'd strongly recommend using a version of HSQL that's patched, and that can work with Java 8 (if one exists, that is).

**Review by mprins — CHANGES_REQUESTED** [review]
You PR suggests CVE-2022-41853 applies to GeoTools, however GeoTools only uses this library for the HSQLDB EPSG module which only uses trusted data (and normally has read-only access)

Since we're in the process of dropping Java 8 on the main branch this upgrade should be considered

File: pom.xml (hunk 1 above)
Comment (pmauduit) [reply pom.xml:1033]: > This is a Java 11 (class file version 55.0) build so will cause failures when used with any Java version before 11.

Sorry, I figured this out after having opened the PR.

Comment (aaime) [reply pom.xml:1033]: We can  merge once we switch main to Java 11... we'll ping you when it's done, so that you can rebase this PR.

Comment (pmauduit) [reply pom.xml:1033]: ok, thanks @aaime

**Comment by aaime** [comment]
Keep an eye on https://github.com/geotools/geotools/pull/4097, after it gets merged you'll be able to rebase and builds should work.

**Comment by viprabhu** [comment]
Would there be a release of geotools that includes hsqldb 2.7.1 and supports Java 8? 
If yes, when can we expect it to be released?

**Comment by mprins** [comment]
@viprabhu No; hsqldb 2.7.1 is Java 11 only so it will never work on java 8

**Comment by viprabhu** [comment]
> @viprabhu No; hsqldb 2.7.1 is Java 11 only so it will never work on java 8
When i look at the website [hsqldb](https://hsqldb.org/) it states it supports JDK 8.

> Latest version 2.7.1 works with JDK 8 and above.

**Comment by mprins** [comment]
> Latest version 2.7.1 works with JDK 8 and above.

@viprabhu It's funny you say that, when looking at this build and comments you can see it failed because hsqldb 2.7.1 has classfile/bytecode version 55.0.
Please backup your statements with a pull request that shows us it works as you say

**Comment by viprabhu** [comment]
> > Latest version 2.7.1 works with JDK 8 and above.
> 
> @viprabhu It's funny you say that, when looking at this build and comments you can see it failed because hsqldb 2.7.1 has classfile/bytecode version 55.0. Please backup your statements with a pull request that shows us it works as you say

I checked the bytecode of hsqldb-2.7.1 binary and it indeed is version 55, looks like the text in the website wrongly states Java 8 support.
`Downloads/hsqldb-2.7.1_jar$ javap -verbose org/hsqldb/jdbc/JDBCDriver.class | grep major`
`  major version: 55`

**Comment by mczap1** [comment]
We are using gt-epsg-hsql version 11.5, we can only use Java 8. We would like to go with mitigation action from github advisory: https://github.com/advisories/GHSA-77xx-rxvh-q682 @mprins can you please advise what should be the value for the property: method_class_names

**Comment by mprins** [comment]
@mczap1 Please do not hijack pull requests with questions, see 
[above](https://github.com/geotools/geotools/pull/4088#pullrequestreview-1159719751)

**Comment by viprabhu** [comment]
> > > Latest version 2.7.1 works with JDK 8 and above.
> > 
> > 
> > @viprabhu It's funny you say that, when looking at this build and comments you can see it failed because hsqldb 2.7.1 has classfile/bytecode version 55.0. Please backup your statements with a pull request that shows us it works as you say
> 
> I checked the bytecode of hsqldb-2.7.1 binary and it indeed is version 55, looks like the text in the official [website ](https://hsqldb.org/) wrongly states Java 8 support for the latest version 2.7.1. `Downloads/hsqldb-2.7.1_jar$ javap -verbose org/hsqldb/jdbc/JDBCDriver.class | grep major` ` major version: 55`

Checked this a bit more deeply and indeed hsqldb 2.7.1 supports Java 8, its published as a separate binary. 
It can be used using maven classifier as `jdk8` in the pom file.
```
        <dependency>
            <groupId>org.hsqldb</groupId>
            <artifactId>hsqldb</artifactId>
            <version>2.7.1</version>
            <classifier>jdk8</classifier>
        </dependency>
```

> Downloads/hsqldb-2.7.1-jdk8$ javap -verbose org/hsqldb/jdbc/JDBCDriver.class | grep major
>   major version: 52

**Comment by aaime** [comment]
@viprabhu nice find
@pmauduit could you update the PR accordingly? It would also allow for backport. The java 11 stuff is moving slowly (no customer/deadline attached to it).

**Comment by viprabhu** [comment]
> @pmauduit could you update the PR accordingly? It would also allow for backport. The java 11 stuff is moving slowly (no customer/deadline attached to it).

I am not sure if this is the right thread to ask this, but based on the above findings in which release of geo-tools supporting Java 8 could we expect hsql 2.7.1 to be included? Could someone help me with some tentative release timeline for it?

**Comment by aaime** [comment]
Superceded by https://github.com/geotools/geotools/pull/4129 (with jdk 8 classifier and a ticket in Jira)

**Comment by pmauduit** [comment]
> Superceded by #4129 (with jdk 8 classifier and a ticket in Jira)

Sorry arriving after the war, I was on vacations with no access to my laptop, thanks for noticing the existence of a classifier for Java8, I haven't stumbled upon it when checking the version on mvnrepository.com.

**Pull request closed without merging**